### DISCUSSION TITLE
Select fields of related entities

### DISCUSSION THREAD

**Discussion by BenStirrup** [ppetzold/nestjs-paginate#903] [Q&A]
Hello 👋 

I am trying to select properties of related entities, for a paginate config such as the following for example : 
```typescript
    {
        relations: ['relation1', 'relation2'],
        select: [
             'id',
             'createdAt',
             'updatedAt',
             'relation1.email',
             'relation1.firstName',
             'relation1.lastName',
         ],
    }
```
The fields from the entity itself can be selected, BUT all fields from the included relations are returned.

I also tried to do the relations' properties select using a query builder passed as param to the `paginate` function, also without success.

Any idea on how to select fields of related entities ? 

Not being able to do that actually has security implications.

**Comment by Helveg** [comment]
As far as security is concerned, FWIW, as a temporary workaround you can `delete` any sensitive keys you don't wish to display from the reponse object yourself. 

If this is broken feel free to open a PR with a test that shows the problem and we can fix your issue :)

I'm also making sure that this feature will work in the next major version according to JSON:API spec! :)
